torch/_dynamo/variables/builtin.py

```diff
@@ -1271,7 +1273,7 @@ def call_type(self, tx, obj: VariableTracker):
             return VariableBuilder(tx, TypeSource(obj.source))(py_type)
 
         if py_type is not None:
-            return ConstantVariable.create(py_type)
+            return SourcelessBuilder()(tx, py_type)
```

Comment:
cc @yanboliang Is this alright? The issue torchvision was hitting was taht  `py_type = torch.Tensor` coming from a sourceless tensor, it was wrongly being wrapped into a ConstantVariable rather than a TorchVariable.

Is this alright, or would you want us to be a bit less general here and just do this for `torch.Tensor` so that most paths still hit the UserError?

Reply:
```VariableTrackers``` produced by ```SourcelessBuilder``` are unguarded, what happens if there is a graph break after a ```type(obj)``` call if ```obj``` is generated internally (doesn't have a source)? I think it works well for constant and torch variables, we can't guarantee if they are other variables. But I don't expect a ```type```call returns too complicated data structure, so it should be fine IMO.

Reply:
Ok, so Ed argues in https://github.com/pytorch/pytorch/pull/113390/ that this is alright (that PR subsumes this one)